interfaces/builtin/docker.go

```diff
@@ -0,0 +1,541 @@
+/{,var/}run/docker.sock rw,
+
+@{PROC}/sys/net/core/somaxconn r,
+`
+
+const dockerPermanentSlotSecComp = `
+# The Docker daemon needs to be able to launch arbitrary processes within
+# containers (whose syscall needs are unknown beforehand)
+
+# Calls the Docker daemon itself requires
+#   /snap/docker/VERSION/bin/docker-runc
+#     "do not inherit the parent's session keyring"
+#     "make session keyring searcheable"
+keyctl
+#   /snap/docker/VERSION/bin/docker-runc
+pivot_root
+
+# This list comes from Docker's default seccomp whitelist (which is applied to
+#   all containers launched unless a custom profile is specified or
+#   "--privileged" is used)
+# https://github.com/docker/docker/blob/v1.12.0/profiles/seccomp/seccomp_default.go#L39-L1879
+# $ grep -C1 'ActAllow' -- profiles/seccomp/seccomp_default.go \
+#     | grep 'Name:' \
+#     | cut -d'"' -f2 \
+#     | sort -u
+accept
+accept4
+access
+acct
+adjtimex
+alarm
+arch_prctl
+bind
+bpf
+breakpoint
+brk
+cacheflush
+capget
+capset
+chdir
+chmod
+chown
+chown32
+chroot
+clock_getres
+clock_gettime
+clock_nanosleep
+clone
+close
+connect
+copy_file_range
+creat
+delete_module
+dup
+dup2
+dup3
+epoll_create
+epoll_create1
+epoll_ctl
+epoll_ctl_old
+epoll_pwait
+epoll_wait
+epoll_wait_old
+eventfd
+eventfd2
+execve
+execveat
+exit
+exit_group
+faccessat
+fadvise64
+fadvise64_64
+fallocate
+fanotify_init
+fanotify_mark
+fchdir
+fchmod
+fchmodat
+fchown
+fchown32
+fchownat
+fcntl
+fcntl64
+fdatasync
+fgetxattr
+finit_module
+flistxattr
+flock
+fork
+fremovexattr
+fsetxattr
+fstat
+fstat64
+fstatat64
+fstatfs
+fstatfs64
+fsync
+ftruncate
+ftruncate64
+futex
+futimesat
+getcpu
+getcwd
+getdents
+getdents64
+getegid
+getegid32
+geteuid
+geteuid32
+getgid
+getgid32
+getgroups
+getgroups32
+getitimer
+getpeername
+getpgid
+getpgrp
+getpid
+getppid
+getpriority
+getrandom
+getresgid
+getresgid32
+getresuid
+getresuid32
+getrlimit
+get_robust_list
+getrusage
+getsid
+getsockname
+getsockopt
+get_thread_area
+gettid
+gettimeofday
+getuid
+getuid32
+getxattr
+init_module
+inotify_add_watch
+inotify_init
+inotify_init1
+inotify_rm_watch
+io_cancel
+ioctl
+io_destroy
+io_getevents
+ioperm
+iopl
+ioprio_get
+ioprio_set
+io_setup
+io_submit
+ipc
+kcmp
+kill
+lchown
+lchown32
+lgetxattr
+link
+linkat
+listen
+listxattr
+llistxattr
+_llseek
+lookup_dcookie
+lremovexattr
+lseek
+lsetxattr
+lstat
+lstat64
+madvise
+memfd_create
+mincore
+mkdir
+mkdirat
+mknod
+mknodat
+mlock
+mlock2
+mlockall
+mmap
+mmap2
+modify_ldt
+mount
+mprotect
+mq_getsetattr
+mq_notify
+mq_open
+mq_timedreceive
+mq_timedsend
+mq_unlink
+mremap
+msgctl
+msgget
+msgrcv
+msgsnd
+msync
+munlock
+munlockall
+munmap
+name_to_handle_at
+nanosleep
+newfstatat
+_newselect
+open
+openat
+open_by_handle_at
+pause
+perf_event_open
+personality
+pipe
+pipe2
+poll
+ppoll
+prctl
+pread64
+preadv
+prlimit64
+process_vm_readv
+process_vm_writev
+pselect6
+ptrace
+pwrite64
+pwritev
+query_module
+read
+readahead
+readlink
+readlinkat
+readv
+reboot
```

Comment:
Looking at https://github.com/docker/docker/blob/master/docs/security/seccomp.md, reboot is not allowed by default. I then compared that list to this list and found that the following are allowed in this policy but not allowed in the docker default policy:

```
acct
adjtimex
bpf
clone
ioperm
iopl
kcmp
lookup_dcookie
mount
perf_event_open
personality
process_vm_readv
process_vm_writev
reboot
setns
settimeofday
umount
umount2
unshare
```

Since you say in this file "This list comes from Docker's default seccomp whitelist" I'd like to better understand these.  First off, I think that the comment should say this instead: "This list comes from Docker's default seccomp whitelist plus syscalls required for docker to properly function." As such, it seems quite clear why clone, mount, setns, umount, umount2 and unshare are needed by docker to funtion. Can you comment on the others?